### PR TITLE
Added support for Customer Collect Mandate

### PR DESCRIPTION
**Pull request by RickJeroen**
Mandate which the debtor can collect with.

### CODE REVIEW

**Comment by willemstuursma** [comment]
Thanks @RickJeroen! Would it be possible for you to add an integration test to https://github.com/php-twinfield/twinfield/blob/master/tests/IntegrationTests/CustomerIntegrationTest.php?

File: src/CustomerCollectMandate.php [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+    }
+
+    /**
+     * @param string $dateString
+     * @return $this
+     * @throws Exception
+     */
+    public function setFirstRunDateFromString(string $dateString): self
+    {
+        if (!empty($dateString)) {
```
Comment (willemstuursma) [inline src/CustomerCollectMandate.php]: I don't think the empty check is necessary if the argument is typehinted to `string`.

File: src/DomDocuments/CustomersDocument.php [hunk 2]
```diff
@@ -103,6 +104,39 @@ public function addCustomer(Customer $customer): void
                 // Add the full element
                 $financialElement->appendChild($element);
             }
+
+            //check if collectmandate should be set
+            if ($customer->getCollectMandate() !== null) {
+
```
Comment (willemstuursma) [inline src/DomDocuments/CustomersDocument.php]: I know it's not all best practice here in this class, but could you try to see if you can use some of the convenience methods in `\PhpTwinfield\DomDocuments\BaseDocument`?

File: src/Mappers/CustomerMapper.php [hunk 3]
```diff
@@ -88,8 +88,34 @@ public static function map(Response $response)
...
+                $collectMandateTags = array(
+                    'id' => 'setID',
+                    'signaturedate' => 'setSignatureDateFromString',
+                    'firstrundate' => 'setFirstRunDateFromString',
+                );
+
+                $customer->setCollectMandate(new \PhpTwinfield\CustomerCollectMandate());
+
+                // Go through each collect mandate element and add to the assigned method
+                foreach ($collectMandateTags as $tag => $method) {
```
Comment (willemstuursma) [inline src/Mappers/CustomerMapper.php]: Any of the convenience methods in `\PhpTwinfield\Mappers\BaseMapper` that can be useful here?

File: src/CustomerCollectMandate.php (hunk 1 above)
Comment (RickJeroen) [reply src/CustomerCollectMandate.php]: The Customer getter may return an empty string for signatureDate and firstRunDate.

Comment (willemstuursma) [reply src/CustomerCollectMandate.php]: Then it should be nullable.

Comment (RickJeroen) [reply src/CustomerCollectMandate.php]: True, just fixed it.

**Pull request merged**